velox/dwio/parquet/writer/Writer.h

```diff
@@ -110,11 +110,13 @@ class Writer : public dwio::common::Writer {
   Writer(
       std::unique_ptr<dwio::common::FileSink> sink,
       const WriterOptions& options,
-      std::shared_ptr<memory::MemoryPool> pool);
+      std::shared_ptr<memory::MemoryPool> pool,
+      RowTypePtr schema = nullptr);
```

Comment:
Let's not allow null schema. It is confusing to use the type from the first batch as the schema for the whole file.

Please, update comments to clarify that schema is required and that it should have noon-empty unique field names.

move 'schema' after 'pool'